Some UT may fail randomly, most because of the expectMsg time out

Exception like this:
```
[info] SplitSpec:
[info] - Split should split the text and deliver to next task *** FAILED ***
[info]   java.lang.AssertionError: assertion failed: timeout (10 seconds) during expectMsg while waiting for Message(this,0)
[info]   at scala.Predef$.assert(Predef.scala:165)
[info]   at akka.testkit.TestKitBase$class.expectMsg_internal(TestKit.scala:338)
[INFO] [01/21/2015 17:00:23.660] [Client$] closing netty client netty-client-HostPort(127.0.0.1,47951)...
[info]   at akka.testkit.TestKitBase$class.expectMsg(TestKit.scala:324)
[info]   at akka.testkit.TestKit.expectMsg(TestKit.scala:718)
[info]   at org.apache.gearpump.streaming.examples.kafka.wordcount.SplitSpec$$anonfun$1$$anonfun$apply$mcV$sp$1$$anonfun$apply$1.apply(SplitSpec.scala:40)
[info]   at org.apache.gearpump.streaming.examples.kafka.wordcount.SplitSpec$$anonfun$1$$anonfun$apply$mcV$sp$1$$anonfun$apply$1.apply(SplitSpec.scala:39)
[info]   at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
[info]   at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
[info]   at org.apache.gearpump.streaming.examples.kafka.wordcount.SplitSpec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(SplitSpec.scala:39)
[info]   at org.apache.gearpump.streaming.examples.kafka.wordcount.SplitSpec$$anonfun$1.apply$mcV$sp(SplitSpec.scala:37
```

--- a/streaming/src/test/scala/org/apache/gearpump/streaming/StreamingTestUtil.scala
+++ b/streaming/src/test/scala/org/apache/gearpump/streaming/StreamingTestUtil.scala
@@ -23,6 +23,7 @@ import akka.actor._
 import akka.dispatch.Dispatcher
 import akka.testkit.{TestProbe, TestActorRef}
 import org.apache.gearpump.Message
+import org.apache.gearpump.cluster.AppMasterToMaster.SaveAppData
 import org.apache.gearpump.cluster.TestUtil.MiniCluster
 import org.apache.gearpump.cluster.master.AppMasterRuntimeInfo
 import org.apache.gearpump.cluster.scheduler.Resource
@@ -30,7 +31,7 @@ import org.apache.gearpump.cluster.{AppMasterContext, UserConfig}
 import org.apache.gearpump.partitioner.HashPartitioner
 import org.apache.gearpump.streaming.AppMasterToExecutor.StartClock
 import org.apache.gearpump.streaming.ExecutorToAppMaster.RegisterTask
-import org.apache.gearpump.streaming.task.{TaskId, StartTime, TaskActor, TaskContext}
+import org.apache.gearpump.streaming.task._
 import org.apache.gearpump.transport.Express
 import org.apache.gearpump.util.Graph
 import org.apache.gearpump.util.Graph._
@@ -75,16 +76,25 @@ object StreamingTestUtil {
     val taskReporter = TestProbe()(system2)
     val taskId1 = TaskId(0, 0)
     val taskId2 = TaskId(1, 0)
-    val appMaster = system1.actorOf(Props(classOf[MockAppMaster]))
+    val appMaster = TestProbe()(system1)
+    def ignoreUpdateClock: PartialFunction[Any, Boolean] = {
+      case msg: UpdateClock => true
+    }
+    appMaster.ignoreMsg(ignoreUpdateClock)
 
     implicit val systemForSerializer = system1.asInstanceOf[ExtendedActorSystem]
 
-    var testActorProps = Props(Class.forName(taskClass), TaskContext(taskId1, 1, 0, appMaster, 1, dag), taskConf)
+    var testActorProps = Props(Class.forName(taskClass), TaskContext(taskId1, 1, 0, appMaster.ref, 1, dag), taskConf)
     if (usePinedDispatcherForTaskActor) {
       testActorProps = testActorProps.withDispatcher("akka.actor.pined-dispatcher")
     }
     val testActor = TestActorRef[TaskActor](testActorProps)(system1)
-    val reporter = system2.actorOf(Props(classOf[EchoTask], TaskContext(taskId2, 2, 0, appMaster, 1, dag), UserConfig.empty.withValue(EchoTask.TEST_PROBE, taskReporter.ref)))
+    appMaster.expectMsgClass(classOf[RegisterTask])
+    appMaster.reply(StartClock(0))
+
+    val reporter = system2.actorOf(Props(classOf[EchoTask], TaskContext(taskId2, 2, 0, appMaster.ref, 1, dag), UserConfig.empty.withValue(EchoTask.TEST_PROBE, taskReporter.ref)))
+    appMaster.expectMsgClass(classOf[RegisterTask])
+    appMaster.reply(StartClock(0))
 
     val express1 = Express(system1)
     val express2 = Express(system2)
@@ -124,11 +134,4 @@ class EchoTask(taskContext : TaskContext, userConf : UserConfig) extends TaskAct
 
 object EchoTask {
   val TEST_PROBE = "test_probe"
-}
-
-class MockAppMaster extends Actor {
-  override def receive: Receive = {
-    case msg: RegisterTask =>
-      sender ! StartClock(0)
-  }
 }

